src/components/media-controls/media-controls.ts

```diff
@@ -60,7 +60,7 @@ export class MediaControlsComponent extends WithShoelace(AbstractComponent(LitEl
 
   /** A DOM selector to target the spectrogram element */
   @property({ type: String })
-  public for = "";
+  public for: string | SpectrogramComponent = "";
```

Comment:
what happens to the attribute when an element reference is assigned? is it removed? is it set to an empty string? does it reflect at all? what should happen?

Reply:
The attribute isn't reflected, so it isn't changed.

You make a good point and I think that it should be reflected.

---

Note that this isn't unique to this change.

If you previously changed the component's `for` property, the attribute wouldn't change, meaning that the attribute could become disassociated with the property.

This might actually be the cause of some perf issues because when a framework like Angular would re-render, it'd use the attributes value and then the property would be updated.

Reply:
![image](https://github.com/user-attachments/assets/9cadbc9b-8858-409f-95a6-910ed506e66b)

Reply:
so what does this actually look like when reflected?

to be clear I think this is a case where we can justifiably expect `for=""`

Reply:
![image](https://github.com/user-attachments/assets/1f62d02d-7284-4438-89d8-14f8e7952eb9)

After I've added the `reflect: true`, it now has the text `for="[object HTMLElement]"`.

As weird as this looks, I think that it's fine. It's correctly showing that it's using an element ref.

---

I think there might be a case for overriding the `SpectrogramComponent`'s `toString()` method so that it returns it's element id, so that if you assign an element reference, the id is reflected in the DOM document.

Reply:
Soni don't really care for this pr, but as a general principle for this repo I think we need a custom reflection serializer.

IMO the best solution is a valid query selector string